lib/model/content.dart

```diff
@@ -1451,6 +1451,61 @@ class _ZulipContentParser {
     return tableNode ?? UnimplementedBlockContentNode(htmlNode: tableElement);
   }
 
+  List<BlockContentNode> parseMathBlocks(dom.Element element) {
+    assert(_debugParserContext == _ParserContext.block);
+    assert(element.localName == 'p');
+    assert((() {
+      final first = element.nodes.first;
+      return first is dom.Element
+        && first.localName == 'span'
+        && first.className == 'katex-display';
+    })());
+    final debugHtmlNode = kDebugMode ? element : null;
+
+    final blocks = <BlockContentNode>[];
```

Comment:
My performance-paranoia for this parsing code doesn't like the idea of allocating this list, just to immediately empty it into the surrounding list, particularly in the common case where there's only going to be a single math block in the `<p>` element. (And here "common case" means "everything but the weird edge case".)

But I think avoiding that would require a larger refactor, so let's go ahead with this way. It's probably fine (that is, it's probably not a big performance impact in practice). And at least it only affects messages with math blocks in them, which isn't a super common feature in the first place.